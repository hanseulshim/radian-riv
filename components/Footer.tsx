import * as React from 'react'

const Footer: React.FC = () => {
  return (
    <div id="footer">
      <img src={`${process.env.baseUrl}/images/footer-logo.svg`} alt="logo" />
      <span>
        © 2020 Red Bell Real Estate, LLC. All rights reserved. Ownership and use
        of this report is governed by the legal agreement between RedBell and
        the party for which it was prepared. Information contained in this
        report is current as at the date of the report. Red Bell does not accept
        any liability for use of this report or the data, all of which is
        provided “as-is” and without warranty or covenant of any kind including
        any warranty as to its accuracy, marketability or fitness for a
        particular purpose or usage. Any use not expressly authorized by such
        agreement, including reliance on this report, is prohibited.
      </span>
    </div>
  )
}

export default Footer
