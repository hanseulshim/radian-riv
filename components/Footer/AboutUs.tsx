import React, { useState, useCallback } from 'react'
import Modal from 'components/common/Modal'
import { getAboutUs } from 'api'
import Table from 'components/common/Table'

interface Props {
  closeModal: () => void
}

export default function AboutUs({ closeModal }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    const tableData = await getAboutUs()
    if (tableData.length) {
      const tableColumns = Object.keys(tableData[0]).map(key => ({
        Header: key,
        accessor: key
      }))
      setColumns(tableColumns)
    }
    setData(tableData)
  }, [])

  const getColWidth = (header: string) => ({
    minWidth: 250
  })

  return (
    <Modal closeModal={closeModal} title={'About Us'} percent={80}>
      <Table
        columns={columns}
        data={data}
        fetchData={fetchData}
        getColWidth={getColWidth}
      />
    </Modal>
  )
}
