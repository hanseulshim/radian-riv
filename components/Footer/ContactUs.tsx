import React, { useState } from 'react'
import Input from 'components/common/Input'
import Modal from 'components/common/Modal'
import Form from 'components/common/Form'
import { validateEmail } from 'utils'

interface Props {
  closeModal: () => void
}

export default function ContactUs({ closeModal }: Props) {
  const toEmail = 'vow@redbellre.com'
  const [name, setName] = useState('')
  const [email, setEmail] = useState('')
  const [subject, setSubject] = useState('')
  const [address, setAddress] = useState('')
  const [comment, setComment] = useState('')
  const [emailError, setEmailError] = useState('')
  const [alert, setAlert] = useState(null)
  const submitForm = async () => {
    setAlert(null)
    const error = validateEmail(email)
    if (error) {
      setEmailError(error)
    } else {
      setAlert({ type: 'success', message: 'Email Sent' })
      const a = document.createElement('a')
      a.href = `mailto:${toEmail}?subject=${subject}&body=Name: ${name}%0D%0AEmail: ${email}%0D%0AProperty Address: ${address}%0D%0AComment: ${comment}`
      a.click()
    }
  }
  return (
    <Modal closeModal={closeModal} title={'Contact Us'} width={800}>
      <Form id="contact-us" onSubmit={submitForm} alert={alert}>
        <Input
          label="To"
          value={toEmail}
          error={''}
          onChange={() => {}}
          disabled
        />
        <div className="form-row">
          <div className="form-spacer">
            <Input
              label="Name"
              value={name}
              error={''}
              onChange={e => setName(e.target.value)}
            />
          </div>
          <div className="form-spacer">
            <Input
              label="Email"
              value={email}
              error={emailError}
              onChange={e => {
                setEmail(e.target.value)
                setEmailError('')
              }}
            />
          </div>
        </div>
        <Input
          label="Subject"
          value={subject}
          error={''}
          onChange={e => setSubject(e.target.value)}
        />
        <Input
          label="Property Address"
          value={address}
          error={''}
          onChange={e => setAddress(e.target.value)}
        />
        <textarea
          placeholder="Comment Field..."
          onChange={e => setComment(e.target.value)}
          value={comment}
        />
        <button className="btn" type="submit">
          Submit
        </button>
      </Form>
    </Modal>
  )
}
