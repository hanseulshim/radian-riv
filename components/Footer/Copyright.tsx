import { useState } from 'react'
import Input from 'components/common/Input'
import Modal from 'components/common/Modal'
import Form from 'components/common/Form'
import { validateEmail } from 'utils'
import copyrightText from './copyrightText'

interface Props {
  closeModal: () => void
}

export default function Copyright({ closeModal }: Props) {
  const toEmail = 'vow@redbellre.com'
  const subject = 'Copyright Infringement'
  const [name, setName] = useState('')
  const [email, setEmail] = useState('')
  const [address, setAddress] = useState('')
  const [comment, setComment] = useState('')
  const [emailError, setEmailError] = useState('')
  const [alert, setAlert] = useState(null)
  const submitForm = async () => {
    setAlert(null)
    const error = validateEmail(email)
    if (error) {
      setEmailError(error)
    } else {
      setAlert({ type: 'success', message: 'Email Sent' })
      const a = document.createElement('a')
      a.href = `mailto:${toEmail}?subject=${subject}&body=Name: ${name}%0D%0AEmail: ${email}%0D%0AProperty Address: ${address}%0D%0AComment: ${comment}`
      a.click()
    }
  }
  return (
    <Modal closeModal={closeModal} title={'Copyright Infringement'} width={800}>
      <div
        className="copyright"
        dangerouslySetInnerHTML={{ __html: copyrightText }}
      />
      <Form id="reset-password" onSubmit={submitForm} alert={alert}>
        <Input
          label="To"
          value={toEmail}
          error={''}
          onChange={() => {}}
          disabled
        />
        <div className="form-row">
          <div className="form-spacer">
            <Input
              label="Name"
              value={name}
              error={''}
              onChange={e => setName(e.target.value)}
            />
          </div>
          <div className="form-spacer">
            <Input
              label="Email"
              value={email}
              error={emailError}
              onChange={e => {
                setEmail(e.target.value)
                setEmailError('')
              }}
            />
          </div>
        </div>
        <Input
          label="Subject"
          value={subject}
          error={''}
          onChange={() => {}}
          disabled
        />
        <Input
          label="Property Address"
          value={address}
          error={''}
          onChange={e => setAddress(e.target.value)}
        />
        <textarea
          placeholder="Comment Field..."
          onChange={e => setComment(e.target.value)}
          value={comment}
        />
        <button className="btn btn-primary" type="submit">
          Submit
        </button>
      </Form>
    </Modal>
  )
}
