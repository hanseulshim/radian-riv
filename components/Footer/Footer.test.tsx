import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Footer from '.'

describe('Footer', () => {
  let component
  beforeAll(() => {
    component = shallow(<Footer />)
  })
  it('has a div with id footer', () => {
    expect(component.find('#footer').prop('id')).toEqual('footer')
  })
  it('has an image logo', () => {
    expect(component.find('img').prop('src')).toContain('footer-logo.svg')
  })
  it('has "About Us" link', () => {
    expect(component.find('a').at(0).text()).toEqual('About Us')
  })
  it('has "Privacy Policy" link', () => {
    expect(component.find('a').at(1).text()).toEqual('Privacy Policy')
    expect(component.find('a').at(1).prop('href')).toContain(
      'PrivacyPolicy.pdf'
    )
  })
  it('has "Terms of Use" link', () => {
    expect(component.find('a').at(2).text()).toEqual('Terms of Use')
    expect(component.find('a').at(2).prop('href')).toContain('TermsOfUse.pdf')
  })
  it('has "Copyright Infringement" link', () => {
    expect(component.find('a').at(3).text()).toEqual('Copyright Infringement')
  })
  it('has "Contact Us" link', () => {
    expect(component.find('a').at(4).text()).toEqual(
      'Contact Us: vox@redbellre.com'
    )
  })
  it('has copyright text', () => {
    expect(component.find('span').text().length).toBeGreaterThan(100)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
