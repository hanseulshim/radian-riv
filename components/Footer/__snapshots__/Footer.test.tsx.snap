// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<div
  id="footer"
>
  <div
    className="logo-container"
  >
    <img
      alt="logo"
      src="undefined/images/footer-logo.svg"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="footer-links"
    >
      <a
        onClick={[Function]}
      >
        About Us
      </a>
      <a
        href="undefined/PrivacyPolicy.pdf"
        target="_blank"
      >
        Privacy Policy
      </a>
      <a
        href="undefined/TermsOfUse.pdf"
        target="_blank"
      >
        Terms of Use
      </a>
      <a
        onClick={[Function]}
      >
        Copyright Infringement
      </a>
      <a
        onClick={[Function]}
      >
        Contact Us: vox@redbellre.com
      </a>
    </div>
    <span>
      © 2020 Red Bell Real Estate, LLC. All rights reserved. Ownership and use of this report is governed by the legal agreement between RedBell and the party for which it was prepared. Information contained in this report is current as at the date of the report. Red Bell does not accept any liability for use of this report or the data, all of which is provided “as-is” and without warranty or covenant of any kind including any warranty as to its accuracy, marketability or fitness for a particular purpose or usage. Any use not expressly authorized by such agreement, including reliance on this report, is prohibited.
    </span>
  </div>
</div>
`;
